enterprise/coderd/proxyhealth/proxyhealth.go

```diff
@@ -0,0 +1,235 @@
+package proxyhealth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/coderd/database/dbauthz"
+)
+
+type ProxyHealthStatus string
+
+const (
+	// Reachable means the proxy access url is reachable and returns a healthy
+	// status code.
+	Reachable ProxyHealthStatus = "reachable"
+	// Unreachable means the proxy access url is not responding.
+	Unreachable ProxyHealthStatus = "unreachable"
```

Comment:
Would be nice to have `OK, Unhealthy, Unreachable, Unregistered` and a reason field.

And make the moon reach out to the primary to test both directions